compiler/rustc_resolve/src/imports.rs

```diff
@@ -509,6 +512,10 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     prev_root_id = import.root_id;
                 }
             } else if is_indeterminate {
+                if let ImportKind::Single { target_bindings, .. } = &import.kind
+                && self.has_resolution(&target_bindings) {
+                    continue;
```

Comment:
I don't understand why it's legal to skip the error here.
What guarantees that the compilation won't be successful?

I also don't understand why `has_resolution` was selected as a criterion for skipping errors.
From what I understand, `import_dummy_binding` above should guarantee that all indeterminate bindings will have resolutions.
So this line just skips errors from *all* indeterminate single imports?

Reply:
Yep, I had concerned it too, I think I should deal to the error output rather than skip errors.

Reply:
> So this line just skips errors from all indeterminate single imports?

No, `import_dummy_binding` only define dummy binding for resolution but not modify the value of `target_bindings`.